tools/scripts/lock-api-state.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -ex
+
+## lock in current protobuf state
+
+# make gen-buf-image ensures that it starts with a clean git state
+make -C proto gen-buf-image
+# check to see if gen-buf-image resulted in any file changes or not
+if [[ -z "$(git status --porcelain)" ]]; then
+    echo "buf image is already up to date"
+    exit 0
+fi
+git add --all
```

Comment:
Ok, so technically we require a clean working directory to run this script, so technically there is no difference between '--all' and '--update', but I think that after the very first run, --update will always be sufficient and --all is an unnecessary risk of committing all untracked files in somebody's repository.

blocking: can we use --update instead?

Reply:
yeah good idea. I didn't know about it --update, thanks!